diff --git a/extra/getopt_long.c b/extra/getopt_long.c
index 8dfda800..d463e332 100644
--- a/extra/getopt_long.c
+++ b/extra/getopt_long.c
@@ -73,7 +73,7 @@ __progname(char * nargv0)
  *     Parse argc/argv argument vector.
  */
 int
-getopt_internal(int nargc, char ** nargv, const char *ostr)
+getopt_internal(int nargc, char * const * nargv, const char *ostr)
 {
        static char *place = EMSG;              /* option letter processing */
        const char *oli;                                /* option letter list index */
