diff --git a/CMakeLists.txt b/CMakeLists.txt
index 49d9a125..ab1bf5ce 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,13 +27,15 @@ endif()

 set(CMAKE_COMPILE_WARNING_AS_ERROR ON CACHE BOOL "Treat compilation warning as error")

-if(NOT MSVC)
+if (NOT CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
   # cmake 3.24 introduces this variable, but for backward compatibility, we set it explicitly
   if (CMAKE_COMPILE_WARNING_AS_ERROR)
       add_definitions(-Werror)
   endif ()

-  add_definitions(-Wall)
+  if (NOT WIN32)
+      add_definitions(-Wall)
+  endif()
   add_definitions(-Wsign-compare)
   add_definitions(-Wconversion)
   add_definitions(-Wno-sign-conversion)
